src/js/index.js

```diff
@@ -48,11 +51,19 @@ if (
   tryRegisterSW();
 } else {
   window.console.warn(
-    `Not trying to register Service Worker because 
+    `Not trying to register Service Worker because
     ENVIRONMENT = ${ENVIRONMENT} and NODE_ENV = ${NODE_ENV}`
   );
 }
 
+if(NODE_ENV === 'test'){
+  const {loadAPI} = require('./mocks/mockapi');
+  const mockjax = require('jquery-mockjax');
+
+  mockjax($, window);
+  loadAPI();
+}
+
```

Comment:
we shouldn't have test code in production code - what's the purpose of this? We should configure the API endpoint used by the app using the environment variable DEFAULT_API_URL - see the README

Reply:
In test environment, I added this to mock the api requests so I can test with selenium, DEFAULT_API_URL is used but it's called inside the `loadAPI` function.

Reply:
What I'm saying is we don't want code like `if(NODE_ENV === 'test'){` in a production module (index.js)

Now I can see that to use mockjax, you need to execute javascript in the client context.

I think for end-to-end tests like there, a mock server would be more appropriate than intercepting ajax requests in jquery.

So you'd spin up a mock server, and build the client with `DEFAULT_API_URL=http://localhost:mockport yarn build` or alternatively open the page with ?promptapi in the URL like http://localhost:8080/services/?promptapi and have selenium fill that in if it can.